src/System.Private.ServiceModel/tools/test/Bridge/WcfTestBridgeCommon/PortManager.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+
+namespace WcfTestBridgeCommon
+{
+    // This class exists to create and delete firewall rules to
+    // manage which ports are open on behalf of the Bridge.
+    public class PortManager
+    {
+        private static HashSet<int> s_inPorts = new HashSet<int>();
+        private static HashSet<int> s_outPorts = new HashSet<int>();
+        private static object s_portLock = new object();
+
+        public static void OpenPortInFirewall(int port)
+        {
+            OpenPortInFirewall(port, "in", s_inPorts);
+            OpenPortInFirewall(port, "out", s_outPorts);
+        }
+
+        private static void OpenPortInFirewall(int port, string direction, HashSet<int> ports)
+        {
+            lock (s_portLock)
```

Comment:
This can be done without calling external processes. See [this blog post](http://blogs.msdn.com/b/securitytools/archive/2009/08/21/automating-windows-firewall-settings-with-c.aspx) for how to do it. Also I did some checking and for http.sys, you add a rule to allow the process with name "system" to receive traffic, for the other transports you can use the bridge process .exe location. I'm okay with this for now, but I'd like to change it to programmatically being changed.